src/classes/HiFiAxisConfiguration.ts

```diff
@@ -4,270 +4,63 @@
-            isValid = false;
-        }
-        if (axisConfiguration.outOfScreenAxis === HiFiAxes.PositiveZ && axisConfiguration.intoScreenAxis !== HiFiAxes.NegativeZ) {
-            HiFiLogger.error(`Invalid axis configuration!\nIntoScreen Axis is ${axisConfiguration.intoScreenAxis}, and OutOfScreen Axis is ${axisConfiguration.outOfScreenAxis}!`);
-            isValid = false;
-        }
-        // END into-screen/out-of-screen axis error checking
-
-        // START up/down axis error checking
-        if (axisConfiguration.upAxis === HiFiAxes.PositiveX && axisConfiguration.downAxis !== HiFiAxes.NegativeX) {
-            HiFiLogger.error(`Invalid axis configuration!\nUp Axis is ${axisConfiguration.upAxis}, and Down Axis is ${axisConfiguration.downAxis}!`);
-            isValid = false;
-        }
-        if (axisConfiguration.downAxis === HiFiAxes.PositiveX && axisConfiguration.upAxis !== HiFiAxes.NegativeX) {
-            HiFiLogger.error(`Invalid axis configuration!\nUp Axis is ${axisConfiguration.upAxis}, and Down Axis is ${axisConfiguration.downAxis}!`);
-            isValid = false;
-        }
-
-        if (axisConfiguration.upAxis === HiFiAxes.PositiveY && axisConfiguration.downAxis !== HiFiAxes.NegativeY) {
-            HiFiLogger.error(`Invalid axis configuration!\nUp Axis is ${axisConfiguration.upAxis}, and Down Axis is ${axisConfiguration.downAxis}!`);
-            isValid = false;
-        }
-        if (axisConfiguration.downAxis === HiFiAxes.PositiveY && axisConfiguration.upAxis !== HiFiAxes.NegativeY) {
-            HiFiLogger.error(`Invalid axis configuration!\nUp Axis is ${axisConfiguration.upAxis}, and Down Axis is ${axisConfiguration.downAxis}!`);
-            isValid = false;
-        }
-
-        if (axisConfiguration.upAxis === HiFiAxes.PositiveZ && axisConfiguration.downAxis !== HiFiAxes.NegativeZ) {
-            HiFiLogger.error(`Invalid axis configuration!\nUp Axis is ${axisConfiguration.upAxis}, and Down Axis is ${axisConfiguration.downAxis}!`);
-            isValid = false;
-        }
-        if (axisConfiguration.downAxis === HiFiAxes.PositiveZ && axisConfiguration.upAxis !== HiFiAxes.NegativeZ) {
-            HiFiLogger.error(`Invalid axis configuration!\nUp Axis is ${axisConfiguration.upAxis}, and Down Axis is ${axisConfiguration.downAxis}!`);
-            isValid = false;
-        }
-        // END up/down axis error checking
-
-        if (!(axisConfiguration.handedness === HiFiHandedness.RightHand || axisConfiguration.handedness === HiFiHandedness.LeftHand)) {
-            HiFiLogger.error(`Invalid axis configuration!\nHandedness is ${axisConfiguration.handedness}!`);
-            isValid = false;
-        }
-
-        return isValid;
-    }
-
-    /** 
-     * ⚠ WARNING ⚠ The code in this function IS wrong.
-     * TODO: implement the function, just a NO OP at the moment.
-     * 
-     * @param axisConfiguration 
-     * @param inputPoint3D 
```

Comment:
The **WorldFrameConfiguration** replaces **HiFiAxisConfiguration** and is simpler. There are only three degrees of freedom when choosing a sensible coordinate frame:

**(1)** forward direction
**(2)** up direction
**(3)** left- or right-handedness

The third "direction" of can be derived.